--- /home/feng/SGE/GE2011.11p1/source/libs/sched/debit.c	2012-05-23 15:04:40.000000000 -0400
+++ source/debit.c	2018-10-12 11:34:36.000000000 -0400
@@ -62,7 +62,7 @@
 
 static int
 debit_job_from_hosts(lListElem *job, lList *granted, lList *host_list, lList *complex_list, 
-                     int *sort_hostlist);
+                     int *sort_hostlist, u_long32 ja_task_id);
 
 static int
 debit_job_from_rqs(lListElem *job, lList *granted, lList *rqs_list, lListElem *pe,
@@ -157,7 +157,8 @@
       if (a->pe) {
          pe_debit_slots(a->pe, a->slots, a->job_id);
       }   
-      debit_job_from_hosts(a->job, a->gdil, a->host_list, a->centry_list, sort_hostlist);
+      /* add ja_task_id for hosts */
+      debit_job_from_hosts(a->job, a->gdil, a->host_list, a->centry_list, sort_hostlist, a->ja_task_id);
       debit_job_from_queues(a->job, a->gdil, a->queue_list, a->centry_list, orders);
       debit_job_from_rqs(a->job, a->gdil, a->rqs_list, a->pe, a->centry_list, a->acl_list, a->hgrp_list);
       debit_job_from_ar(a->job, a->gdil, a->ar_list, a->centry_list);
@@ -265,7 +266,8 @@
 lList *granted,     /* JG_Type */
 lList *host_list,   /* EH_Type */
 lList *centry_list, /* CE_Type */
-int *sort_hostlist
+int *sort_hostlist,
+u_long32 ja_task_id
 ) {
    lSortOrder *so = NULL;
    lListElem *gel, *hep;
@@ -301,8 +303,8 @@
          lSetUlong(hep, EH_load_correction_factor, ulc_factor);
       }   
 
-      debit_host_consumable(job, host_list_locate(host_list, "global"), centry_list, slots, master_task);
-      debit_host_consumable(job, hep, centry_list, slots, master_task);
+      debit_host_consumable(job, host_list_locate(host_list, "global"), centry_list, slots, master_task, ja_task_id);
+      debit_host_consumable(job, hep, centry_list, slots, master_task, ja_task_id);
       master_task = false;
 
       /* compute new combined load for this host and put it into the host */
@@ -334,12 +336,13 @@
  * centry_list: CE_Type
  */
 int 
-debit_host_consumable(lListElem *jep, lListElem *hep, lList *centry_list, int slots, bool is_master_task) 
+debit_host_consumable(lListElem *jep, lListElem *hep, lList *centry_list, int slots, bool is_master_task, u_long32 job_task_id) 
 {
+/* add job_task_id FZ*/
    return rc_debit_consumable(jep, hep, centry_list, slots, 
                            EH_consumable_config_list, 
                            EH_resource_utilization, 
-                           lGetHost(hep, EH_name), is_master_task);
+                           lGetHost(hep, EH_name), is_master_task, job_task_id);
 }
 
 /****** sge_resource_quota_schedd/debit_job_from_rqs() **********************************
